modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java

```diff
@@ -153,21 +172,66 @@ public static com.liferay.object.model.ObjectField toObjectField(
 			LocalizedMapUtil.getLocalizedMap(objectField.getLabel()));
 		serviceBuilderObjectField.setName(objectField.getName());
 		serviceBuilderObjectField.setObjectFieldSettings(
-			TransformUtil.transformToList(
-				objectField.getObjectFieldSettings(),
-				objectFieldSetting ->
-					ObjectFieldSettingUtil.toObjectFieldSetting(
-						objectField.getBusinessTypeAsString(),
-						objectFieldSetting, objectFieldSettingLocalService,
-						objectFilterLocalService)));
+			Collections.emptyList());
 		serviceBuilderObjectField.setRequired(
 			GetterUtil.getBoolean(objectField.getRequired()));
+		serviceBuilderObjectField.setState(objectField.getState());
 		serviceBuilderObjectField.setSystem(
 			GetterUtil.getBoolean(objectField.getSystem()));
 
 		return serviceBuilderObjectField;
 	}
 
+	private static void _addListTypeEntries(
+			ListTypeDefinition listTypeDefinition,
+			ListTypeEntryLocalService listTypeEntryLocalService,
+			ObjectField objectField, long userId)
+		throws Exception {
+
+		for (ObjectFieldSetting objectFieldSetting :
+				objectField.getObjectFieldSettings()) {
+
+			if (!StringUtil.equals(
+					objectFieldSetting.getName(),
+					ObjectFieldSettingConstants.NAME_STATE_FLOW)) {
+
+				continue;
+			}
+
+			Map<String, Object> valuesMap =
+				(Map<String, Object>)objectFieldSetting.getValue();
+
+			Object objectStates = valuesMap.get("objectStates");
+
+			List<Object> values = null;
+
+			if (objectStates instanceof Object[]) {
+				values = ListUtil.fromArray((Object[])objectStates);
+			}
+			else {
+				values = (List<Object>)objectStates;
+			}
+
+			for (Object value : values) {
+				HashMap<String, Object> objectState =
+					(HashMap<String, Object>)value;
+
+				String key = (String)objectState.get("key");
+
+				ListTypeEntry listTypeEntry =
+					listTypeEntryLocalService.fetchListTypeEntry(
+						listTypeDefinition.getListTypeDefinitionId(), key);
+
+				if (listTypeEntry == null) {
+					listTypeEntryLocalService.addListTypeEntry(
+						userId, listTypeDefinition.getListTypeDefinitionId(),
+						key,
+						Collections.singletonMap(LocaleUtil.getDefault(), key));
+				}
+			}
```

Comment:
@murilostodolni could you please check if we can simplify this code using `JSONObject` and `JSONArray`, as follows:

```
JSONObject jsonObject = JSONFactoryUtil.createJSONObject(
	JSONFactoryUtil.looseSerializeDeep(
		objectFieldSetting.getValue()));

JSONArray objectStatesJSONArray = jsonObject.getJSONArray("objectStates");

for (int i = 0; i < objectStatesJSONArray.length(); i++) {
	JSONObject objectStateJSONObject = objectStatesJSONArray.getJSONObject(i);

	String key = objectStateJSONObject.getString("key");

	ListTypeEntry listTypeEntry =
			listTypeEntryLocalService.fetchListTypeEntry(
				listTypeDefinition.getListTypeDefinitionId(), key);

	if (listTypeEntry == null) {
		listTypeEntryLocalService.addListTypeEntry(
			userId, listTypeDefinition.getListTypeDefinitionId(),
			key,
			Collections.singletonMap(LocaleUtil.getDefault(), key));
	}
}
```